src/server/sample_data/routes/lib/load_data.js

```diff
@@ -23,75 +23,77 @@ import zlib from 'zlib';
 
 const BULK_INSERT_SIZE = 500;
 
-export function loadData(path, bulkInsert, callback) {
-  let count = 0;
-  let docs = [];
-  let isPaused = false;
+export function loadData(path, bulkInsert) {
+  return new Promise(function (resolve, reject) {
```

Comment:
Only change to this file is that instead of using the callback pattern, the function now returns a promise so it is easier to call from within a loop